docs/extensions/bastion.md

```diff
@@ -0,0 +1,42 @@
+# Contract: `Bastion` resource
```

Comment:
Can you add this doc to the index (https://github.com/gardener/gardener/blob/master/docs/README.md#extensions)?